jormungandr-lib/src/interfaces/vote/mod.rs

```diff
@@ -0,0 +1,69 @@
+use serde::Serialize;
+
+#[derive(Serialize)]
+#[serde(remote = "chain_impl_mockchain::block::BlockDate")]
+struct BlockDate {
+    pub epoch: u32,
+    pub slot_id: u32,
+}
+
+#[derive(Serialize)]
+#[serde(remote = "chain_impl_mockchain::certificate::ExternalProposalId")]
+struct ExternalProposalId {
+    #[serde(getter = "chain_impl_mockchain::certificate::ExternalProposalId::to_string")]
+    id: String,
+}
+
+#[derive(Serialize)]
+pub struct Proposal {
+    pub external_id: String,
+    pub options: u8,
+}
```

Comment:
what prevents us from implementing the Remote here?

Reply:
I couldn't chain the base method `external_id` and `to_string` for external_id and the `options` with `as_byte` for options. I'm gonna try making an specific funtion to mine that data.